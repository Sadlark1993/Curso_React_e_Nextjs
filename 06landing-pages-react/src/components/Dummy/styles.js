import styled, { css } from 'styled-components';

export const titleStyle = styled.div`
  ${({ theme }) => css``}
`;
