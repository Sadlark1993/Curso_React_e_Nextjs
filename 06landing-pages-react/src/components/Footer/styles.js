import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;
    a {
      color: inherit;
      font-size: ${theme.fonts.size.small};
      text-align: center;
    }
  `}
`;
