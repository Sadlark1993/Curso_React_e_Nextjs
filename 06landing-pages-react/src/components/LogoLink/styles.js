import styled, { css } from 'styled-components';

export const LogoStyle = styled.a`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    color: inherit;

    & > img {
      max-height: 2.5rem;
    }
  `}
`;
