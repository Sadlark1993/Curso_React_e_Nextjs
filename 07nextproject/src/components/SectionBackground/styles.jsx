import styled, { css } from 'styled-components';

const activateBg = (theme) => css`
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`;

export const bgStyle = styled.div`
  ${({ theme, background }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};

    ${background && activateBg(theme)}
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
