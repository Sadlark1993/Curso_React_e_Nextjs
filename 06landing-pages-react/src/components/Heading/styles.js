import styled, { css } from 'styled-components';
import { theme } from '../../styles/themes';

export const titleStyle = styled.h1`
  ${({ theme, lightT }) => css`
    color: ${lightT ? theme.colors.white : theme.colors.primaryColor};
  `}
`;
