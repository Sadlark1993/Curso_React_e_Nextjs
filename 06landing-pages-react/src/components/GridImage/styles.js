import styled, { css } from 'styled-components';
import { titleStyle } from '../Heading/styles';
import { Container as TextContainer } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    & > ${TextContainer} {
      margin-bottom: ${theme.spacing.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacing.large};
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    transition: all 300ms ease-in-out;

    &:hover {
      transform: scale(1.2) rotate(12deg);
    }
  `}
`;
