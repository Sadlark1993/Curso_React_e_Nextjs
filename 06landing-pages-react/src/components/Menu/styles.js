import styled, { css } from 'styled-components';
import { ContainerStyle } from '../SectionContainer/styles';
import { titleStyle } from '../Heading/styles';

export const MenuStyle = styled.div`
  ${({ theme }) => css`
    position: fixed;
    z-index: 5;
    top: 0;
    left: 0;
    right: 0;
    border-bottom: ${theme.colors.mediumGray};
    background: ${theme.colors.white};

    /* Changing css properties of the ContainerStyle when it's child of MenuStyle */
    > ${ContainerStyle} {
      padding-top: 0;
      padding-bottom: 0;
    }

    & ${titleStyle} {
      margin-top: 0;
      margin-bottom: 0;
    }

    @media ${theme.media.mobile} {
      height: 100vh;

      /* Changing css properties of the ContainerStyle when it's child of MenuStyle */
      > ${ContainerStyle} {
        display: grid;
        grid-template-columns: 1fr;
        grid-template-rows: 1fr;
        height: 100vh;
        align-items: center;
        overflow-y: auto;
      }

      & ${titleStyle} {
        padding-bottom: ${theme.spacing.large};
        display: flex;
        justify-content: center;
      }
    }
  `}
`;

export const MenuStyle2 = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;

    @media ${theme.media.mobile} {
      display: block;
      text-align: center;
      padding: ${theme.spacing.medium};
    }
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css``}
`;
