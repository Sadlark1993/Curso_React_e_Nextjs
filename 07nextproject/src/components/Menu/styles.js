import styled, { css } from 'styled-components';
import { ContainerStyle } from '../SectionContainer/styles';
import { titleStyle } from '../Heading/styles';

const menuVisible = (theme) => css`
  visibility: visible;
  opacity: 1;
`;

export const MenuStyle = styled.div`
  ${({ theme, visible }) => css`
    position: fixed;
    z-index: 5;
    top: 0;
    left: 0;
    right: 0;
    border-bottom: ${theme.colors.mediumGray};
    background: ${theme.colors.white};
    transition: all 300ms ease-in-out;

    /* Changing css properties of the ContainerStyle when it's child of MenuStyle */
    > ${ContainerStyle} {
      padding-top: 0;
      padding-bottom: 0;
    }

    & ${titleStyle} {
      margin-top: 0;
      margin-bottom: 0;
    }

    @media ${theme.media.mobile} {
      height: 100vh;
      /* same as display:none, except for the fact that with this, we can use transition. */
      visibility: hidden;
      opacity: 0;
      ${visible && menuVisible(theme)};

      /* Changing css properties of the ContainerStyle when it's child of MenuStyle */
      > ${ContainerStyle} {
        display: grid;
        grid-template-columns: 1fr;
        grid-template-rows: 1fr;
        height: 100vh;
        align-items: center;
        overflow-y: auto;
      }

      & ${titleStyle} {
        padding-bottom: ${theme.spacing.large};
        display: flex;
        justify-content: center;
      }
    }
  `}
`;

export const MenuStyle2 = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;

    @media ${theme.media.mobile} {
      display: block;
      text-align: center;
      padding: ${theme.spacing.medium};
    }
  `}
`;

export const Button = styled.button`
  ${({ theme, visible }) => css`
    z-index: 6;
    position: fixed;
    top: 2rem;
    right: 2rem;
    width: 4rem;
    height: 4rem;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    border: none;
    display: none;
    /* if menu is visible, this block don't captures the pointer actions. */
    pointer-events: ${visible ? 'none' : 'all'};

    @media ${theme.media.mobile} {
      display: flex;
      align-items: center;
      justify-content: center;
    }

    > svg {
      width: 2.5rem;
      height: 2.5rem;
    }
  `}
`;
