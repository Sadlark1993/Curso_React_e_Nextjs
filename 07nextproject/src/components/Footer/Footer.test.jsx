import { renderTheme } from '../../styles/render-theme';
import { Footer } from '.';
import { screen } from '@testing-library/react';

describe('<Footer />', () => {
  it('Should render Footer', () => {
    const { container } = renderTheme(<Footer html={'<h1>Ol√°</h1>'} />);
    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
