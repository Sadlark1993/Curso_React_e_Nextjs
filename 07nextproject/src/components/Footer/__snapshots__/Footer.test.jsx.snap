// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Should render Footer 1`] = `
.c4 {
  font-size: 2.4rem;
}

.c2 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  border-top: solid 0.2rem #DDDDDD;
  text-align: center;
}

.c0 a {
  color: inherit;
  text-align: center;
}

.c0 .c3 {
  font-size: 1.6rem;
}

.c0 .c1 {
  padding-top: 0;
  padding-bottom: 0;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <div
        class="c3 c4"
      >
        <h1>
          Ol√°
        </h1>
      </div>
    </div>
  </div>
</div>
`;
